Tests/tests/test_MotionDetection.cpp

```diff
@@ -96,107 +95,212 @@ TEST_F(MotionDetectionEventTest, getMotionDetectors)
+    EXPECT_EQ(response,  string(""));
 }
 
 TEST_F(MotionDetectionEventTest, setSensitivity)
 {
+    ON_CALL(motionDetectionImplMock, MOTION_DETECTION_SetSensitivity(::testing::_,::testing::_,::testing::_))
+        .WillByDefault(::testing::Invoke(
+            [](std::string index, std::string sensitivity, int inferredMode) {
+                return MOTION_DETECTION_RESULT_SUCCESS;
+            }));
+
 
-        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("setSensitivity"), _T("{\"index\":\"FP_MD\",\"name\":\"high\",\"value\":\"2\"}"), response));
-        EXPECT_EQ(response,  string("{\"success\":true}"));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("setSensitivity"), _T("{\"index\":\"FP_MD\",\"name\":\"high\",\"value\":\"40\"}"), response));
+    EXPECT_EQ(response,  string("{\"success\":true}"));
+}
+
+TEST_F(MotionDetectionEventTest, setSensitivityInvalid)
+{
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("setSensitivity"), _T("{\"index\":\"FP_MD\"}"), response));
+    EXPECT_EQ(response,  string(""));
 }
 
 TEST_F(MotionDetectionEventTest, getSensitivity)
 {
-          ON_CALL(motionDetectionImplMock, MOTION_DETECTION_GetSensitivity(::testing::_,::testing::_,::testing::_))
+    ON_CALL(motionDetectionImplMock, MOTION_DETECTION_GetSensitivity(::testing::_,::testing::_,::testing::_))
         .WillByDefault(::testing::Invoke(
             [](std::string index, char** sensitivity, int* currentMode) {
                 *currentMode = 1;
-                *sensitivity = (char *)malloc(sizeof(STR_SENSITIVITY_LOW));
-                memset(*sensitivity, 0, sizeof(STR_SENSITIVITY_LOW));
-                strcpy(*sensitivity, STR_SENSITIVITY_LOW);
+                *sensitivity = (char *)malloc(sizeof(STR_SENSITIVITY_HIGH));
+                memset(*sensitivity, 0, sizeof(STR_SENSITIVITY_HIGH));
+                strcpy(*sensitivity, STR_SENSITIVITY_HIGH);
                 return MOTION_DETECTION_RESULT_SUCCESS;
             }));
 
-        EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getSensitivity"), _T("{\"index\":\"FP_MD\"}"), response));
-        EXPECT_EQ(response,  string("{\"value\":\"high\",\"success\":true}"));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("getSensitivity"), _T("{\"index\":\"FP_MD\"}"), response));
+    EXPECT_EQ(response,  string("{\"value\":\"high\",\"success\":true}"));
+}
+
+TEST_F(MotionDetectionEventTest, getSensitivityInvalid)
+{
+    ON_CALL(motionDetectionImplMock, MOTION_DETECTION_GetSensitivity(::testing::_,::testing::_,::testing::_))
+        .WillByDefault(::testing::Invoke(
+            [](std::string index, char** sensitivity, int* currentMode) {
+                return MOTION_DETECTION_RESULT_INDEX_ERROR;
+            }));
+
+    EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("getSensitivity"), _T("{}"), response));
```

Comment:
is it returning ERROR_GENERAL due to invalid parameters or due to MOTION_DETECTION_RESULT_INDEX_ERROR? This should be more clear from the test or its name